src/main/java/uk/gov/hmcts/reform/hmc/service/PendingRequestServiceImpl.java

```diff
@@ -0,0 +1,151 @@
+                hearingId
+            );
+        }
+        return lockedRequests;
+    }
+
+    public List<PendingRequestEntity> findQueuedPendingRequestsForProcessing() {
+        List<PendingRequestEntity> pendingRequests =
+            pendingRequestRepository.findQueuedPendingRequestsForProcessing(
+                getIntervalUnits(pendingWaitInterval), getIntervalMeasure(pendingWaitInterval));
+        if (!pendingRequests.isEmpty()) {
+            pendingRequests.forEach(e ->
+                log.debug("findQueuedPendingRequestsForProcessing(): id<{}> hearingId<{}> ",
+                      e.getId(), e.getHearingId()));
+        } else {
+            log.debug("findQueuedPendingRequestsForProcessing(): empty");
+        }
+        return pendingRequests;
+    }
+
+    public void markRequestAsPending(Long id, Integer retryCountIn, LocalDateTime lastTriedDateTimeIn) {
+        log.debug("markRequestAsPending({}, {}, {})", id, retryCountIn, lastTriedDateTimeIn);
+        int retryCountOut = retryCountIn + 1;
+        LocalDateTime lastTriedDateTimeOut = LocalDateTime.now();
+        pendingRequestRepository.markRequestAsPending(id, retryCountOut, lastTriedDateTimeOut);
+        log.debug("markRequestAsPending({}, {}, {})", id, retryCountOut, lastTriedDateTimeOut);
+    }
+
+    public void markRequestWithGivenStatus(Long id, String status) {
+        log.debug("markRequestWithGivenStatus({}, {})", id, status);
+        pendingRequestRepository.markRequestWithGivenStatus(id, status);
+    }
+
+    public void escalatePendingRequests() {
+        log.debug("escalatePendingRequests()");
+
+        try {
+            List<PendingRequestEntity> pendingRequests =
+                pendingRequestRepository.findRequestsForEscalation(getIntervalUnits(escalationWaitInterval),
+                                                                   getIntervalMeasure(escalationWaitInterval));
+            pendingRequests.forEach(this::escalatePendingRequest);
+        } catch (Exception e) {
+            log.error("Failed to escalate Pending Requests");
+        }
+
+    }
+
+    public void deleteCompletedPendingRequests() {
+        log.debug("deleteCompletedPendingRequests({})", deletionWaitInterval);
+        try {
+            int countOfDeletedRecords = pendingRequestRepository.deleteCompletedRecords(
+                getIntervalUnits(deletionWaitInterval), getIntervalMeasure(deletionWaitInterval));
+            log.debug("{} Completed pendingRequests deleted", countOfDeletedRecords);
+        } catch (Exception e) {
+            log.error("Failed to deleteCompletedRecords");
+        }
+    }
+
+    protected void escalatePendingRequest(PendingRequestEntity pendingRequest) {
+        log.debug("escalatePendingRequests");
+        pendingRequestRepository.markRequestForEscalation(pendingRequest.getId(), LocalDateTime.now());
+
+        log.error("Error occurred for pending request. Service:{}. Entity:{}. Method:{}. Hearing ID:{}.",
```

Comment:
```suggestion
        log.error("Error occurred for pending request. Service:{}. Entity:{}. Method:{}. Hearing id:{}.",
```

Reply:
The regex for dynatrace to pick up the error and create a ticket is on log.error and message contains Hearing id so just make sure we match case sensitivity